app/src/main/java/org/stepik/android/adaptive/pdd/ui/dialog/StreakRestoreDialog.kt

```diff
@@ -0,0 +1,64 @@
+package org.stepik.android.adaptive.pdd.ui.dialog
+
+import android.app.Activity
+import android.app.Dialog
+import android.content.Intent
+import android.os.Bundle
+import android.support.v4.app.DialogFragment
+import android.support.v7.app.AlertDialog
+import org.stepik.android.adaptive.pdd.R
+import org.stepik.android.adaptive.pdd.data.AnalyticMgr
+import org.stepik.android.adaptive.pdd.databinding.DialogStreakRestoreBinding
+import org.stepik.android.adaptive.pdd.ui.fragment.CardsFragment
+import org.stepik.android.adaptive.pdd.util.InventoryUtil
+
+class StreakRestoreDialog : DialogFragment() {
+    companion object {
+        private val STREAK_KEY = "streak"
+
+        fun newInstance(streak: Long) : StreakRestoreDialog {
+            AnalyticMgr.getInstance().onStreakRestoreDialogShown()
+            val dialog = StreakRestoreDialog()
+            dialog.arguments = Bundle()
+            dialog.arguments.putLong(STREAK_KEY, streak)
+            return dialog
+        }
+    }
+
+    private lateinit var binding : DialogStreakRestoreBinding
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val alertDialogBuilder = AlertDialog.Builder(context, R.style.ExpLevelDialogTheme)
+        binding = DialogStreakRestoreBinding.inflate(activity.layoutInflater, null, false)
+
+        binding.ticketItem.counter.text = getString(R.string.amount, InventoryUtil.getItemsCount(InventoryUtil.TICKETS_KEY))
+
+        binding.useCouponButton.setOnClickListener {
+            if (InventoryUtil.useItem(InventoryUtil.TICKETS_KEY)) {
+                binding.ticketItem.counter.text = getString(R.string.amount, InventoryUtil.getItemsCount(InventoryUtil.TICKETS_KEY))
+                onStreakRestore()
+            }
+            dismiss()
+        }
+
+        binding.cancelButton.setOnClickListener {
+            AnalyticMgr.getInstance().onStreakRestoreCanceled(arguments?.getLong(STREAK_KEY) ?: 0)
+            dismiss()
+        }
+
+        alertDialogBuilder.setView(binding.root)
+
+        val dg = alertDialogBuilder.create()
+        dg.setCancelable(false)
+        dg.setCanceledOnTouchOutside(false)
```

Comment:
I have checked source code of appcompat:
```
        public AlertDialog create() {
            // We can't use Dialog's 3-arg constructor with the createThemeContextWrapper param,
            // so we always have to re-set the theme
            final AlertDialog dialog = new AlertDialog(P.mContext, mTheme);
            P.apply(dialog.mAlert);
            dialog.setCancelable(P.mCancelable);
            if (P.mCancelable) {
                dialog.setCanceledOnTouchOutside(true);
            }
            dialog.setOnCancelListener(P.mOnCancelListener);
            dialog.setOnDismissListener(P.mOnDismissListener);
            if (P.mOnKeyListener != null) {
                dialog.setOnKeyListener(P.mOnKeyListener);
            }
            return dialog;
        }
```

`P.mCancelable` is flag from builder, and dialog will be created with `setCanceledOnTouchOutside(true)` automatically.